### DISCUSSION TITLE
File upload from flutter

### DISCUSSION THREAD

**Discussion by Akshaykakkodi** [pocketbase/pocketbase#4678] [Q&A] [answered]
How to upload files from flutter app

**Comment by ganigeorgiev** [comment]
You can check the Dart example from the ["Uploading files" docs](https://pocketbase.io/docs/files-handling/#uploading-files).

There is also an example in the ["File upload" section](https://github.com/pocketbase/dart-sdk#file-upload) in the Dart SDK readme.

If the above doesn't help, please elaborate what is not clear in the above and what platform you are targeting (web, desktop, mobile).

**Reply by ganigeorgiev** [reply]
For the available `http.MultipartFile` constructors and methods you can refer to the standard Dart http docs - https://pub.dev/documentation/http/latest/http/MultipartFile-class.html.

**Comment by Akshaykakkodi** [comment]
Future<String> selectImageFromGallery() async {
  XFile? file = await ImagePicker()
      .pickImage(source: ImageSource.gallery, imageQuality: 10);
  if (file != null) {
    return file.path;
  } else {
    return '';
  }
}


 selectLogoFromCamera()async{
    selectedLogoPath=await coreController.selectImageFromCamera();
    notifyListeners();
  }


image picker is used to pick images . how to send this image as multipart files

**Reply by ganigeorgiev** [reply]
You can retrieve the file content as bytes list using [`XFile.readAsBytes()`](https://pub.dev/documentation/image_picker/latest/image_picker/XFile/readAsBytes.html) and load it into `http.MultipartFile` using the [`MultipartFile.fromBytes()` constructor](https://pub.dev/documentation/http/latest/http/MultipartFile/MultipartFile.fromBytes.html). Something like:

```dart
final raw = await file.readAsBytes();

final multipartFile = http.MultipartFile.fromBytes('yourFileField', raw, filename: file.name);

...
```

**Reply by ganigeorgiev** [reply]
Another alternative could be to pass directly the stream generated with  [`XFile.openRead()`](https://pub.dev/documentation/image_picker/latest/image_picker/XFile/openRead.html) to the default [`MultipartFile()` constructor](https://pub.dev/documentation/http/latest/http/MultipartFile/MultipartFile.html) but to use that you'll need first to await the [XFile.length()](https://pub.dev/documentation/image_picker/latest/image_picker/XFile/length.html). Something like:

```dart
final length = await file.length();
final fileStream = file.openRead();

final multipartFile = http.MultipartFile('yourFileField', fileStream, length, filename: file.name);

...
```

I'm not sure which one would be more efficient but the bytes one seems more simple.

**Reply by Akshaykakkodi** [reply]
thank you , i will try both and update the result

**Reply by Akshaykakkodi** [reply, marked as answer]
files: [
        http.MultipartFile.fromBytes(
            'logo',
            
            File(selectedLogoPath).readAsBytesSync(),
            filename: selectedLogoPath,
        ),
      
    ]);

this worked

**Reply by ganigeorgiev** [reply]
If you have access to the file path you can use directly the [`await http.MultipartFile.fromPath()`](https://pub.dev/documentation/http/latest/http/MultipartFile/fromPath.html) factory method but note that using a path is platform-dependent and may not work on Flutter web or other platform that doesn't have dart:io or similar.
